### PR TITLE
Remove 404 response from rootDataStoreRequestHandler

### LINKED ISSUE #8723: rootDataStoreRequestHandler should return undefined instead of create404Response()
When `rootDataStoreRequestHandler` is passed as a request handler to `buildRuntimeRequestHandler()` and it fails to execute, it returns an error `create404Response()`. This doesn't allow the other request handlers passed after `rootDataStoreRequestHandler` to run. 

Instead, `rootDataStoreRequestHandler` should return `undefined` and the next request handler should get executed.

### PR DESCRIPTION
**Pull request by sonalivdeshpande**
Resolves: https://github.com/microsoft/FluidFramework/issues/8723

### CODE REVIEW

File: packages/framework/request-handler/src/test/requestHandlers.spec.ts [hunk 1]
```diff
@@ -92,7 +92,8 @@ describe("RequestParser", () => {
         it("Data store request with sub route", async () => {
             const requestParser = RequestParser.create({ url: "/objectId/route", headers: { wait: true } });
             const response = await rootDataStoreRequestHandler(requestParser, runtime);
-            assert.equal(response.status, 200);
+            console.log("response--------", response);
```
Comment (ChumpChief) [inline packages/framework/request-handler/src/test/requestHandlers.spec.ts:95]: Remove log before checkin

**Review by ChumpChief — APPROVED** [review]

File: packages/framework/request-handler/src/requestHandlers.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const rootDataStoreRequestHandler = async (request: IRequest, runtime: IC
         const rootDataStore = await runtime.getRootDataStore(id, wait);
         return rootDataStore.IFluidRouter.request(requestParser.createSubRequest(1));
     } catch (error) {
-        return create404Response(request);
```
Comment (anthony-murphy) [inline packages/framework/request-handler/src/requestHandlers.ts:43]: this doesn't seem right. it should only return undefined if there in no root data store

Comment (anthony-murphy) [reply packages/framework/request-handler/src/requestHandlers.ts:43]: i think getRootDataStore return undefined if the id doesn't exist, that's the only case we should return undefined. in the error case we probably want to return a 500.

Comment (anthony-murphy) [reply packages/framework/request-handler/src/requestHandlers.ts:43]: uggh. looked at the code. looks like getRootDataStore fails if the data store does not exist. but to try catching getRootDataStore and returning undefined is ok, but we shouldn't wrap `rootDataStore.IFluidRouter.request(requestParser.createSubRequest(1));` as if that throws, the request was to a valid datastore, it just failed to process it. in that case we should return 500

Comment (sdeshpande3) [reply packages/framework/request-handler/src/requestHandlers.ts:43]: yep. I'll have a PR out for this change.

Comment (ssimic2) [reply packages/framework/request-handler/src/requestHandlers.ts:43]: Agree with @anthony-murphy here on using 500. I'd leave 404 only for builder as a means to say "I tried all the handlers, none of them resolved."

Comment (sdeshpande3) [reply packages/framework/request-handler/src/requestHandlers.ts:43]: try/catch PR for `rootDataStoreRequestHandler`: https://github.com/microsoft/FluidFramework/pull/8742

**Pull request merged**